python/dataproc_templates/gcs/gcs_to_spanner.py

```diff
@@ -0,0 +1,160 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Sequence, Optional, Any
+from logging import Logger
+import argparse
+import pprint
+
+from pyspark.sql import SparkSession, DataFrame
+
+from dataproc_templates import BaseTemplate
+import dataproc_templates.util.template_constants as constants
+
+__all__ = ['GCSToSpannerTemplate']
+
+
+class GCSToSpannerTemplate(BaseTemplate):
+  """
+  Dataproc template implementing loads from GCS into Spanner
+  """
+
+  @staticmethod
+  def parse_args(args: Optional[Sequence[str]] = None) -> Dict[str, Any]:
+    parser: argparse.ArgumentParser = argparse.ArgumentParser()
+
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_INPUT_LOCATION}',
+      dest=constants.GCS_SPANNER_INPUT_LOCATION,
+      required=True,
+      help='GCS location of the input files'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_INPUT_FORMAT}',
+      dest=constants.GCS_SPANNER_INPUT_FORMAT,
+      required=True,
+      # TODO: Add support to CSV?
+      help='Input file format (one of: <avro,parquet,orc>)',
+      choices=[
+        constants.FORMAT_AVRO,
+        constants.FORMAT_PRQT,
+        constants.FORMAT_ORC,
+      ]
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_INSTANCE}',
+      dest=constants.GCS_SPANNER_OUTPUT_INSTANCE,
+      required=True,
+      help='Spanner output instance'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_DATABASE}',
+      dest=constants.GCS_SPANNER_OUTPUT_DATABASE,
+      required=True,
+      help='Spanner output database'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_TABLE}',
+      dest=constants.GCS_SPANNER_OUTPUT_TABLE,
+      required=True,
+      help='Spanner output table'
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_PRIMARY_KEY}',
+      dest=constants.GCS_SPANNER_PRIMARY_KEY,
+      required=False,
+      help=(
+        "Primary key of the table. "
+        "It is only required when the table doesn't exist"
+      )
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_SAVE_MODE}',
+      dest=constants.GCS_SPANNER_OUTPUT_SAVE_MODE,
+      required=False,
+      default=constants.OUTPUT_MODE_APPEND,
+      help=(
+        'Output write mode '
+        '(one of: <Overwrite|ErrorIfExists|Append|Ignore>) '
+        '(Defaults to ErrorIfExists)'
+      ),
+      choices=[
+        constants.OUTPUT_MODE_OVERWRITE,
+        constants.OUTPUT_MODE_APPEND,
+        constants.OUTPUT_MODE_IGNORE,
+        constants.OUTPUT_MODE_ERRORIFEXISTS
+      ]
+    )
+    parser.add_argument(
+      f'--{constants.GCS_SPANNER_OUTPUT_BATCH_INSERT_SIZE}',
+      dest=constants.GCS_SPANNER_OUTPUT_BATCH_INSERT_SIZE,
+      required=False,
+      default=1000,
+      help='Spanner batch insert size '
```

Comment:
The default batch size should be documented in the help and the README, so that users know the default value without needing to jump into the code

Reply:
Good catch. Done in 42fabea